apps/content-gateway-api/src/app/maintenance/Maintainer.ts

```diff
@@ -0,0 +1,24 @@
+import { MaintenanceJob } from "./MaintenanceJob";
+import { ToadScheduler } from "toad-scheduler";
+import { pipe } from "fp-ts/lib/function";
+import { makeToToadJob } from ".";
+import { Logger } from "tslog";
+
+export interface Maintainer {
+    readonly jobs: MaintenanceJob[];
+    stop: () => void;
+}
+
+export const attachJobs = (
+    jobs: MaintenanceJob[],
+    scheduler: ToadScheduler,
+    logger: Logger
+) => {
+    const toToadJob = makeToToadJob(logger);
+    jobs.forEach((job) => {
+        pipe(job, toToadJob, function (j) {
+            // We need a wrapper function here to preserve scheduler's this value
```

Comment:
This `this` business is very annoying I agree. 😒

Reply:
Yeah it tripped me up pretty good for a second 😂